TestProject/output/BaseOutput.php

```diff
@@ -0,0 +1,52 @@
+<?php
+namespace output;
+
+use options\Getopt;
+use cellularAutomat\Board;
+
+/**
+ * BaseOutput class which provides the functions for pluggable-output.
+ *
+ * @class BasOutput
+ */
+class BaseOutput
+{
+    /**
+     * @var int creates a new image*.png after every run.
+     */
+    protected $imageIndex = 0;
+
+    /**
+     * Provides additional options.
+     *
+     * @param Getopt $_options
+     */
+    function addOptions(Getopt $_options)
+    {
+    }
+
+    /**
+     * Calls a single file.
+     *
+     * @param Getopt $_options
+     */
+    function startOutput(Getopt $_options)
+    {
+    }
+
+    /**
+     * Prints out the board.
```

Comment:
"Prints" is wrong here. An output might print, another one might send en email per generation, or whatever. Use something general.